File: 프로그래머스/lv2/42883. 큰 수 만들기/큰 수 만들기.js
function solution(number, k) {
    const stack = [];
    const numbers = [...number].map(Number)
    
    let cnt = k;
    for (let num of numbers) {
        if (!stack.length) {
            stack.push(num)
            continue;
        }
        if (cnt > 0) {
            while (stack[stack.length - 1] < num) {
                stack.pop();
                cnt--;
                if (stack.length === 0 || cnt <= 0) break;
            }
        } 
        stack.push(num);
    }
    return stack.splice(0, number.length - k).join('')
}

File: 프로그래머스/lv2/42883. 큰 수 만들기/큰 수 만들기.py
def solution(number, k):
    stack = []
    
    i = 0
    while i < len(number):
        n = int(number[i])
        if stack:
            while stack and k and stack[-1] < n:
                stack.pop()
                k -= 1
        stack.append(n)
        i += 1
    return ''.join(list(map(str, stack[:len(stack) - k])))